test/app/App/Panel/AddToken/index.test.js

```diff
@@ -35,14 +35,31 @@ jest.mock('../../../../../resources/link')
 
 const AddToken = Restore.connect(AddTokenComponent, store)
 const user = userEvent.setup()
+let rerenderComponent
 
 beforeEach(() => {
   link.invoke.mockImplementation((channel, contractAddress, chainId) => {
     if (contractAddress === '0x3432b6a60d23ca0dfca7761b7ab56459d9c964d0') {
       return Promise.resolve(chainMetadata[chainId])
     }
 
-    return Promise.resolve({})
+    return Promise.resolve({ name: '', symbol: '', decimals: 0 })
+  })
+
+  link.send.mockImplementation((channel, msg, { view, data: { notify, notifyData } }) => {
+    expect(channel).toBe('tray:action')
+    expect(msg).toBe('navDash')
+    expect(view).toBe('tokens')
+    expect(notify).toBe('addToken')
+    rerenderComponent(
```

Comment:
why do we need an explicit re-render here? changes in state should drive the rendering of any component and if we change the state in the test that should cause the re-render.

Reply:
navigation updates props not state.

Reply:
I meant the state of the component not "React state". updating the props should still cause the component to re-render

Reply:
Ok, so I wouldn't have put this in if it was behaving as you describe, spent a few hours trying different things but could not get the rendered element to update in the tests until I tried with rerender.  Perhaps testing the individual screen components in isolation will work, that's going to balloon the test file size and the time spent on it but I can certainly give that a go.

https://testing-library.com/docs/example-update-props/
https://testing-library.com/docs/react-testing-library/api/#rerender

Reply:
my concern is that if it's not re-rendering for some reason that may also happen in production. if I understand the code correctly the act of entering data should cause a re-render already, and if it's not that means there could be something wrong w. the code, not just the test.

Reply:
actually I think I see what you're saying now, I didn't understand the way the nav was driving stuff.

Reply:
Yeah I thought it was a minor change but adding nav support to an existing state-driven component is a fair bit of work and it changes the tests substantially too.  We should get better at building components for the nav in the first instance over time, and also at testing them.